tests/test_de_page.py

```diff
@@ -0,0 +1,213 @@
+# Tests for parsing a page from the German Wiktionary
+
+import unittest
+
+from collections import defaultdict
+
+from wikitextprocessor import Wtp
+
+from wiktextract.config import WiktionaryConfig
+from wiktextract.extractor.de.page import (
+    parse_page,
+    parse_section,
+    fix_level_hierarchy_of_subsections,
+)
+from wiktextract.thesaurus import close_thesaurus_db
+from wiktextract.wxr_context import WiktextractContext
+
+
+class DePageTests(unittest.TestCase):
+    def setUp(self):
+        conf1 = WiktionaryConfig(
+            dump_file_lang_code="de",
+            # capture_language_codes=None,
+            # capture_translations=True,
+            # capture_pronunciation=True,
+            # capture_linkages=True,
+            # capture_compounds=True,
+            # capture_redirects=True,
+            # capture_examples=True,
+        )
+        self.wxr = WiktextractContext(Wtp(lang_code="de"), conf1)
+
+    def tearDown(self) -> None:
+        self.wxr.wtp.close_db_conn()
+        close_thesaurus_db(
+            self.wxr.thesaurus_db_path, self.wxr.thesaurus_db_conn
+        )
+
+    def test_de_parse_page(self):
+        self.wxr.wtp.add_page("Vorlage:Sprache", 10, "")
+        lst = parse_page(
+            self.wxr,
+            "Beispiel",
+            """
+== Beispiel ({{Sprache|Deutsch}}) ==
+""",
+        )
+        self.assertEqual(
+            lst,
+            [
+                {
+                    "lang": "Deutsch",
+                    "lang_code": "de",
+                    "word": "Beispiel",
+                }
+            ],
+        )
+
+    def test_de_parse_page_skipping_head_templates(self):
+        self.wxr.wtp.add_page("Vorlage:Wort der Woche", 10, "")
+        self.wxr.wtp.add_page("Vorlage:Siehe auch", 10, "")
+        self.wxr.wtp.add_page("Vorlage:Sprache", 10, "")
+        lst = parse_page(
+            self.wxr,
+            "Beispiel",
+            """
+{{Wort der Woche|46|2020}}
+{{Siehe auch|[[cát]]}}
+== Beispiel ({{Sprache|Deutsch}}) ==
+""",
+        )
+        self.assertEqual(
+            lst,
+            [
+                {
+                    "lang": "Deutsch",
+                    "lang_code": "de",
+                    "word": "Beispiel",
+                }
+            ],
+        )
+
+    # The way append_base_data() works requires the presence of a sense
+    # dictionary before starting a new pos section. Therefore, we need to add
+    # at least one sense data point to the test case.
+    def test_de_parse_section(self):
+        self.wxr.wtp.add_page("Vorlage:Wortart", 10, "")
+        self.wxr.wtp.add_page("Vorlage:Bedeutungen", 10, "")
+        page_text = """
+=== {{Wortart|Adjektiv|Englisch}}, {{Wortart|Adverb|Englisch}} ===
+{{Bedeutungen}}
+:[1] gloss1
+=== {{Wortart|Verb|Englisch}} ===
+{{Bedeutungen}}
+:[1] gloss1
```

Comment:
The data here should be different from data elsewhere, otherwise your tests might pass if there's cross-contamination; if glossAdverb is accidentally used for Verb, when it should be glossVerb, this doesn't catch that. So gloss1, gloss1, gloss1 > gloss1, gloss2, gloss3, etc.